static_frame/core/index_hierarchy.py

```diff
@@ -953,19 +1001,19 @@ def _ufunc_set(self,
                 index_constructors=index_constructors,
                 depth_reference=self.depth)
 
-
     #---------------------------------------------------------------------------
+
+    @property
+    def _index_constructors(self) -> tp.Tuple[tp.Type[Index], ...]:
+        return tuple(index.__class__ for index in self._indices)
```

Comment:
You might have this be an iterator of `tp.Type[Index]` so that the caller can decide if they need a tuple or not.